internal/k8s/k8s.go

```diff
@@ -128,13 +146,118 @@ func (c *Client) ApplyManifests(ctx context.Context, manifests []string) error {
 				})
 
 				if retryErr != nil {
-					return fmt.Errorf("error updating resource: %w", retryErr)
+					return fmt.Errorf("error updating resource: %w ", retryErr)
 				}
 			}
 
-			return fmt.Errorf("error creating resource: %w", err)
+			return fmt.Errorf("error creating resource: %w ", err)
 		}
 	}
 
 	return nil
 }
+
+// WaitForDeploymentReady waits for a target Deployment to become ready
+func (c *Client) WaitForDeploymentReady(ctx context.Context, deployment *appsv1.Deployment, timeoutSeconds int) (bool, error) {
+	log := logger.New(logger.Debug)
+	deploymentName := deployment.Name
+	namespace := deployment.Namespace
+
+	// Get the desired number of replicas from the deployment spec
+	if deployment.Spec.Replicas == nil {
+		return false, fmt.Errorf("deployment %s in Namespace %s has nil Spec.Replicas", deploymentName, namespace)
+	}
+	desiredReplicas := *deployment.Spec.Replicas
+
+	log.Info(fmt.Sprintf("waiting for deployment %q in namespace %q to be ready - this could take up to %d seconds", deploymentName, namespace, timeoutSeconds))
+
+	err := wait.PollUntilContextTimeout(ctx, 5*time.Second, time.Duration(timeoutSeconds)*time.Second, true, func(ctx context.Context) (bool, error) {
+		// Get the latest Deployment object
+		currentDeployment, err := c.clientSet.AppsV1().Deployments(namespace).Get(ctx, deploymentName, metav1.GetOptions{})
+		if err != nil {
+			// If we couldn't connect, retry
+			if isNetworkingError(err) {
+				log.Warn("connection error, retrying: %w ", err)
```

Comment:
😆 I see what you did here. However `%w` doesn't work on anything other than error "handling". This is sending it to a logger, not handling the error so I would be really surprised if `%w` is supported.

```suggestion
				log.Warn("connection error, retrying: %s", err)
```